### PR TITLE
Issue 54 - Added Contributor License Agreement

### PR DESCRIPTION
**Pull request by jyanev**
Based on Micronaut's. Still need to implement using cla assistant

### CODE REVIEW

**Review by ruthvikm — APPROVED** [review]

**Review by kate-holdener — DISMISSED** [review]
I think this PR doesn't fully resolve issue #54 . In the description of the issue, I see that "Completion of this issue also includes directions on how to fill out the CLA in the "Contributing to the Library" section of the docs". This part is missing. Also, when looking at Micronaut's CLA on their website, I am able to digitally sign the CLA. The solution provided in this PR is just the wording of the CLA, but other connecting pieces are missing.

**Review by ruthvikm — CHANGES_REQUESTED** [review]
Use [https://cla-assistant.io/]Cla assistant website to configure CLA for Pi4Micronaut. You can update the documentation a bit more precise about Pi4Micronaut and paste the cla documentation which you have used in the CLA.md file while congifuration.

**Comment by SwindleA** [comment]
@ruthvikm  The table of contents is missing. I was able to re-add it but the resulting format is a bit janky. 


image

When you scroll down, the png disappears to the top. 

image



The code to add it back:
```
= Pi4Micronaut Documentation
:source-highlighter: rouge
:rouge-style: thankful_eyes
:stylesheet: style.css
:docinfo: shared
:icons: font
:doctype: book
:notitle:
:toc: left
:toclevels: 3
:sectanchors:
:sectnums:

image::../../../../Pi4Micronaut_logo.png[]

toc::[]
```
This is the first 16 lines. I did not commit any changes, because I am not sure what layout you want.

**Review by ruthvikm — APPROVED** [review]
Made necessary changes

File: micronautpi4j-utils/src/docs/asciidoc/contributingToLibrary.adoc [hunk 1]
```diff
@@ -0,0 +1,83 @@
+== Contribute to the Pi4Micronaut Library
```
Comment (SwindleA) [inline micronautpi4j-utils/src/docs/asciidoc/contributingToLibrary.adoc:1]: Duplicate file

File: micronautpi4j-utils/src/docs/asciidoc/contribute/newComponent.adoc [hunk 2]
```diff
@@ -1,4 +1,4 @@
-=== How to Create a New Component
```
Comment (SwindleA) [inline micronautpi4j-utils/src/docs/asciidoc/contribute/newComponent.adoc:1]: Readd equal sign to make this a subsection of contributing to library

File: micronautpi4j-utils/src/docs/asciidoc/contribute/contributingToLibrary.adoc [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+11. Engage with the Community
+
+** Attend community meetings or join chat groups.
+** Help other contributors or users when you can.
+
+** Note: While your contribution is highly valued, there's no guarantee that all pull requests will be merged. It depends on the library's direction, quality of the contribution, and decisions of the maintainers.
+
+Thanks for considering a contribution to the Pi4Micronaut library! Your involvement helps make the project better for everyone.
+
+include::newComponent.adoc[]
```
Comment (SwindleA) [inline micronautpi4j-utils/src/docs/asciidoc/contribute/contributingToLibrary.adoc:80]: Move this to index.adoc

**Review by SwindleA — CHANGES_REQUESTED** [review]

File: micronautpi4j-utils/src/docs/asciidoc/contributingToLibrary.adoc (hunk 1 above)
Comment (ruthvikm) [reply micronautpi4j-utils/src/docs/asciidoc/contributingToLibrary.adoc]: deleted

File: micronautpi4j-utils/src/docs/asciidoc/contribute/newComponent.adoc (hunk 2 above)
Comment (ruthvikm) [reply micronautpi4j-utils/src/docs/asciidoc/contribute/newComponent.adoc]: made changes

File: micronautpi4j-utils/src/docs/asciidoc/contribute/contributingToLibrary.adoc (hunk 3 above)
Comment (ruthvikm) [reply micronautpi4j-utils/src/docs/asciidoc/contribute/contributingToLibrary.adoc]: moved it to index.adoc

**Review by SwindleA — APPROVED** [review]
Looks good

**Pull request merged**